app/main/posts/detail/post-detail-data.directive.js

```diff
@@ -0,0 +1,253 @@
+module.exports = PostDetailData;
+
+PostDetailData.$inject = [];
+function PostDetailData() {
+    return {
+        restrict: 'E',
+        replace: true,
+        scope: {
+            filters: '=',
+            isLoading: '=',
+            post: '='
+        },
+        controller: PostDetailDataController,
+        template: require('./post-detail-data.html')
+    };
+}
+
+PostDetailDataController.$inject = [
+    '$scope',
+    '$rootScope',
+    '$translate',
+    '$q',
+    '$filter',
+    '$location',
+    'PostEndpoint',
+    'CollectionEndpoint',
+    'UserEndpoint',
+    'TagEndpoint',
+    'FormAttributeEndpoint',
+    'FormStageEndpoint',
+    'FormEndpoint',
+    'Maps',
+    '_',
+    'Notify',
+    'moment',
+    'PostSurveyService'
+    ];
+
+function PostDetailDataController(
```

Comment:
Like you pointed out we should look at refactoring this into a shared controller or moving logic elsewhere. But for now it's fine. Can you make an issue to revisit it next week?

Reply:
https://github.com/ushahidi/platform/issues/2116